### DISCUSSION TITLE
Doesn’t work on Vercel

### DISCUSSION THREAD

**Discussion by ggukggukee** [gladly-team/next-firebase-auth#676] [General]
Hello!
My app is working fine but on Vercel I get this error:
`errorInfo: {
    code: 'app/no-app',
    message: 'The default Firebase app does not exist. Make sure you call initializeApp() before using any of the Firebase services.'
  },`
I tried different versions of firebase and firebase-admin but nothing helps

**Comment by kmjennison** [comment]
Possibly related: #614

**Reply by ggukggukee** [reply]
Changed my auth from `next-auth` to `next-firebase-auth` and because of this error can't push my app to production

**Reply by ggukggukee** [reply]
Error occurred with `firebase-admin` when tried create a custom token. I made it work initializing `firebase-admin` with their library.

**Reply by Mrjuanblack** [reply]
@ggukggukee How did you init `firebase-admin`? I'm initializing it before calling the `inithAuth` method from `next-firebase-auth` and I commented the admin credentials that the initAuth method needs, but I still encounter the same error as you.

**Reply by ggukggukee** [reply]
@Mrjuanblack I used firebase's library for that.
Created firebase init function:
```js
import admin from 'firebase-admin';

function formatFirebasePrivateKey(key) {
  // return key.replace(/\\n/g, '\n');
  return key ? JSON.parse(key) : undefined;
}

export function createFirebaseAdminApp(params) {
  const privateKey = formatFirebasePrivateKey(params.privateKey);

  // if already created, return the same instance
  if (admin.apps.length > 0) {
    return admin.app();
  }

  // create certificate
  const cert = admin.credential.cert({
    projectId: params.projectId,
    clientEmail: params.clientEmail,
    privateKey,
  });

  // initialize admin app
  return admin.initializeApp({
    credential: cert,
    projectId: params.projectId,
    storageBucket: params.storageBucket,
  });
}
```

and then used in where i firebase admin is needed
```js

async function initializeAdmin() {
  const params = {
    projectId: process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID,
    clientEmail: process.env.FIREBASE_CLIENT_EMAIL,
    privateKey: process.env.FIREBASE_PRIVATE_KEY,
  };

  return createFirebaseAdminApp(params);
}
```
for example before creating custom token
```js
await initializeAdmin();

const token = await auth().createCustomToken(user.uuid);
```

**Comment by thanhtutzaw** [comment]
@kmjennison Hi I also get error on vercel deploy.
initAuth
![Screenshot_20240205-215443](https://github.com/gladly-team/next-firebase-auth/assets/71011043/d91ea0b5-584c-41ba-bf99-b7631f9e9611)
